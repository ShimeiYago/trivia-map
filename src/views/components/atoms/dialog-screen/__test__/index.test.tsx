import { shallow, ShallowWrapper } from 'enzyme';
import { DialogScreen, Props } from '..';

let wrapper: ShallowWrapper<Props, unknown, DialogScreen>;

const props: Props = {
  children: 'message',
  theme: 'black',
  position: 'top',
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    wrapper = shallow(<DialogScreen {...props} />);
  });

  it('basic', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('white theme', () => {
    wrapper.setProps({ theme: 'white' });
    expect(wrapper).toMatchSnapshot();
  });

  it('position bottom', () => {
    wrapper.setProps({ position: 'bottom' });
    expect(wrapper).toMatchSnapshot();
  });
});
