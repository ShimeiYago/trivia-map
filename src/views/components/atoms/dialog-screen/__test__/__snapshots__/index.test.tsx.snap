// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "backgroundColor": "rgba(0,0,0,0.8)",
      "borderRadius": 2,
      "color": "white",
      "display": "inline-block",
      "left": "50%",
      "p": 2,
      "position": "absolute",
      "top": "10%",
      "transform": "translate(-50%,-50%)",
      "width": "80%",
      "zIndex": 1001,
    }
  }
>
  message
</ForwardRef(Box)>
`;

exports[`Shallow Snapshot Tests position bottom 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "backgroundColor": "rgba(0,0,0,0.8)",
      "borderRadius": 2,
      "color": "white",
      "display": "inline-block",
      "left": "50%",
      "p": 2,
      "position": "absolute",
      "top": "90%",
      "transform": "translate(-50%,-50%)",
      "width": "80%",
      "zIndex": 1001,
    }
  }
>
  message
</ForwardRef(Box)>
`;

exports[`Shallow Snapshot Tests position freely 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "backgroundColor": "rgba(0,0,0,0.8)",
      "borderRadius": 2,
      "color": "white",
      "display": "inline-block",
      "left": "50%",
      "p": 2,
      "position": "absolute",
      "top": "50%",
      "transform": "translate(-50%,-50%)",
      "width": "80%",
      "zIndex": 1001,
    }
  }
>
  message
</ForwardRef(Box)>
`;

exports[`Shallow Snapshot Tests white theme 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "backgroundColor": "rgba(255,255,255,0.8)",
      "borderRadius": 2,
      "color": "black",
      "display": "inline-block",
      "left": "50%",
      "p": 2,
      "position": "absolute",
      "top": "10%",
      "transform": "translate(-50%,-50%)",
      "width": "80%",
      "zIndex": 1001,
    }
  }
>
  message
</ForwardRef(Box)>
`;
